src/content/blog/2024-10-03-eslint-json-markdown-support.md

```diff
@@ -0,0 +1,110 @@
+        language: "json/json",
+        rules: {
+            "json/no-duplicate-keys": "error",
+        },
+    },
+];
+```
+
+There are several rules built-in to the plugin, and we're looking for more suggestions. Have an idea for a rule? [Open an issue](https://github.com/eslint/json/issues).
+
+For more information on configuring JSON linting, please check out the [README]([`@eslint/json`](https://npmjs.com/package/@eslint/json)).
+
+## Markdown linting with `@eslint/markdown`
+
+Markdown linting is accomplished using the
+[`@eslint/json`](https://npmjs.com/package/@eslint/markdown) plugin, which is the next generation of the [`eslint-plugin-markdown`](https://npmjs.com/package/eslint-plugin-markdown) plugin that only contained a processor. This new officially supported language plugin features parsing and rules for CommonMark and GitHub-Flavored Markdown. To use the plugin, install it from npm:
+
+```shell
+npm install @eslint/markdown -D
+```
+
+Then update your configuration file:
+
+```js
+// eslint.config.js
+import markdown from "@eslint/markdown";
+
+export default [
+    files: ["**/*.md"],
+    plugins: {
+        markdown
+    },
+    language: "markdown/commonmark",
+    rules: {
+        "markdown/no-html": "error"
+    }
+];
+```
+
+Similar to the JSON plugin, there are several rules built-in to the plugin, and we're looking for more suggestions. Have an idea for a rule? [Open an issue](https://github.com/eslint/markdown/issues).
+
+For more information on configuring Markdown linting, please check out the [README]([`@eslint/markdown`](https://npmjs.com/package/@eslint/markdown)).
+
+## Creating custom rules and using Code Explorer
+
+Both `@eslint/json` and `@eslint/markdown` parse source code into an AST and then traverse the AST to run rules, just like ESLint does with JavaScript. That means you can write custom rules for JSON and Markdown just like you can for JavaScript. The primary difference is in the AST format. JSON uses the [Momoa AST](https://github.com/humanwhocodes/momoa/) while Markdown uses [mdast](https://github.com/syntax-tree/mdast). Because there aren't a lot of resources out there to help you inspect alternate AST formats, we're happy to announce the launch of the [Code Explorer](https://explorer.eslint.org).
+
+[![ESLint Code Explorer](/assets/images/blog/2024/code-explorer.png)](https://explorer.eslint.org)
+
+Code Explorer allows you to view and explore the AST of different languages to help in creating custom rules. For JSON and Markdown, you'll get an expandable view of the AST, while JavaScript allows you to see not just the AST but also the scope and code path information that ESLint generates as it parses your code. Going forward, we'll continue to update Code Explorer with new languages and capabilities designed to help you create custom rules.
+
+And the best part? Code explorer is [open source](https://github.com/eslint/code-explorer), so you can help us make it even better.
```

Comment:
```suggestion
And the best part? [Code explorer](https://github.com/eslint/code-explorer) is open source, so you can help us make it even better.
```
Can we add a hyperlink to the Code Explorer text instead of linking to the open-source text?

Reply:
I already linked "Code Explorer" to explorer.eslint.org above. I think it's better not to link the same text to two different URLs.